networkit/cpp/graph/Graph.cpp

```diff
@@ -622,14 +630,61 @@ void Graph::removeEdge(node u, node v) {
     if (isLoop)
         storedNumberOfSelfLoops--;
 
+    // remove edge for source node
     erase<node>(u, vi, outEdges);
     if (weighted) {
         erase<edgeweight>(u, vi, outEdgeWeights);
     }
     if (edgesIndexed) {
         erase<edgeid>(u, vi, outEdgeIds);
     }
+    if (!directed && !isLoop) {
+        // also remove edge for target node
+        erase<node>(v, ui, outEdges);
+        if (weighted) {
+            erase<edgeweight>(v, ui, outEdgeWeights);
+        }
+        if (edgesIndexed) {
+            erase<edgeid>(v, ui, outEdgeIds);
+        }
+    }
+    if (maintainSortedEdges) {
+        // initial index of deleted edge, also represents current index
+        index cur = vi;
 
+        // sort edges of source node from deleted index upwards
+        while (cur < outEdges[u].size() && outEdges[u][cur] > outEdges[u][cur + 1]) {
```

Comment:
It is interesting, that the error only pops up for Windows debug builds. After closer inspection, the following fixes the issue:

```
while (cur + 1< outEdges[u].size() && outEdges[u][cur] > outEdges[u][cur + 1])
```